File: api/admin/deploy-homepage.js
// Admin endpoint to deploy homepage to Shopify
// This uses the Shopify Admin API to create the homepage and set it as the main page

import { readFileSync } from 'fs';
import { join } from 'path';

export default async function handler(req, res) {
  // Only allow POST requests
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    // Get credentials from environment or request body
    const accessToken = req.body.accessToken || process.env.SHOPIFY_APP_ADMIN_API_KEY || process.env.SHOPIFY_ACCESS_TOKEN;
    const shop = req.body.shop || process.env.SHOPIFY_STORE_DOMAIN || '8k5mna-5e.myshopify.com';

    if (!accessToken) {
      return res.status(400).json({
        error: 'Access token required',
        message: 'Please provide accessToken in request body or set SHOPIFY_ACCESS_TOKEN env var'
      });
    }

    console.log('🏠 Deploying homepage to shop:', shop);

    // Read the homepage Liquid template
    const homepageTemplate = readFileSync(
      join(process.cwd(), 'shopify-homepage.liquid'),
      'utf-8'
    );

    // Create page using Shopify Admin GraphQL API
    const createPageMutation = `
      mutation createPage($input: PageCreateInput!) {
        pageCreate(page: $input) {
          page {
            id
            title
            handle
            body
          }
          userErrors {
            field
            message
          }
        }
      }
    `;

    const pageInput = {
      title: 'Welcome to All Dogs Rock',
      handle: 'home',
      body: homepageTemplate,
      published: true,
    };

    // Call Shopify GraphQL API
    const shopifyUrl = `https://${shop}/admin/api/2024-01/graphql.json`;

    const pageResponse = await fetch(shopifyUrl, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'X-Shopify-Access-Token': accessToken,
      },
      body: JSON.stringify({
        query: createPageMutation,
        variables: { input: pageInput }
      }),
    });

    const pageData = await pageResponse.json();

    if (pageData.errors) {
      console.error('❌ GraphQL errors:', pageData.errors);
      return res.status(400).json({
        error: 'Failed to create page',
        details: pageData.errors
      });
    }

    if (pageData.data?.pageCreate?.userErrors?.length > 0) {
      console.error('❌ User errors:', pageData.data.pageCreate.userErrors);
      return res.status(400).json({
        error: 'Failed to create page',
        details: pageData.data.pageCreate.userErrors
      });
    }

    const createdPage = pageData.data.pageCreate.page;
    console.log('✅ Homepage page created:', createdPage.handle);

    return res.status(200).json({
      success: true,
      message: 'Homepage deployed successfully!',
      page: createdPage,
      pageUrl: `https://www.alldogsrockshop.com/pages/${createdPage.handle}`,
      nextSteps: [
        'Go to: Online Store → Themes → Customize',
        'Click: Homepage section',
        'Add section → Custom Liquid or Page',
        'Select the "Welcome to All Dogs Rock" page',
        'Save!'
      ]
    });

  } catch (error) {
    console.error('❌ Deployment error:', error);
    return res.status(500).json({
      error: 'Deployment failed',
      details: error.message
    });
  }
}

File: deploy-homepage.py
#!/usr/bin/env python3
import json
import os
import sys
import http.client

# Read the homepage content
with open('/home/user/all-dogs-rock-api/shopify-homepage.liquid', 'r') as f:
    homepage_html = f.read()

# Prepare the data
data = {
    "page": {
        "title": "Welcome to All Dogs Rock",
        "body_html": homepage_html,
        "published": True
    }
}

json_data = json.dumps(data)

# API credentials
api_key = os.environ.get('SHOPIFY_APP_ADMIN_API_KEY')
if not api_key:
    print('❌ SHOPIFY_APP_ADMIN_API_KEY not found in environment')
    sys.exit(1)

print('🏠 Creating homepage page in Shopify...')
print('')

# Make the API request
conn = http.client.HTTPSConnection("8k5mna-5e.myshopify.com")
headers = {
    'Content-Type': 'application/json',
    'X-Shopify-Access-Token': api_key
}

try:
    conn.request("POST", "/admin/api/2024-01/pages.json", json_data, headers)
    response = conn.getresponse()
    response_data = response.read().decode()

    if response.status in [200, 201]:
        result = json.loads(response_data)
        if 'page' in result and 'id' in result['page']:
            page_id = result['page']['id']
            page_handle = result['page'].get('handle', 'unknown')

            print('✅ SUCCESS! Homepage page created!')
            print('')
            print(f'Page ID: {page_id}')
            print(f'Page Handle: {page_handle}')
            print('')
            print(f'View at: https://www.alldogsrockshop.com/pages/{page_handle}')
            print(f'Edit at: https://admin.shopify.com/store/8k5mna-5e/pages/{page_id}')
            print('')
            print('🎯 Next: Set this as your homepage in Theme Customizer!')
            print('   Go to: Online Store → Themes → Customize')
            print('   Homepage → Add section → Custom Liquid or Page')
            print(f'   Select: "{page_handle}"')
        else:
            print('✅ Page created but unexpected response format')
            print(response_data)
    else:
        print(f'❌ Error: Status {response.status}')
        print(response_data)

except Exception as e:
    print(f'❌ Error: {e}')
finally:
    conn.close()
